Move from from os-common

chaitu236 opened the pull request:
Moved ni-utils related changes from os-common

@ni/rtos

chaitu236 committed:
packagegroup-ni-base: Add ni-utils

Moved status_led init script to ni-utils and added it to packagegroup-ni-base.
Removed all led init code from initscripts-nilrt

Signed-off-by: Chaitanya Vadrevu <chaitanya.vadrevu@ni.com>

chaitu236 committed:
ni-utils: Move nisetbootmode from os-common

Move changes in nisetbootmode from os-common into ni-utils
Remove nisetbootmode from initscripts-nilrt

Signed-off-by: Chaitanya Vadrevu <chaitanya.vadrevu@ni.com>

shruthi-ravi commented on recipes-ni/ni-utils/ni-utils_1.0.bb:
```diff
@@ -5,15 +5,16 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda
-	   file://nisetbootmode \
+           file://nisetbootmode.functions \
+           file://nisetbootmode \
+           file://nisetled \
```
Can you explain why you are moving nisetled to this package?

chaitu236 replied:
nisetled is the init script for status_led which is provided by this package - so moved it here.

shruthi-ravi replied:
I liked the initscripts only package separation from before. The status_led is called by other scripts too directly, so it makes sense that is installed by ni-utils. Can you make initscripts depend on ni-utils instead?

chaitu236 replied:
I get your point, but in my view there is a fundamental difference between initscript and other scripts calling status_led. If this initscript were to do other things in addition to calling status_led, it'd make sense to put it in initscripts, but the sole purpose of this initscript is to make sure status_led runs at boot time, so I feel like it should go in same package. We always ship initscripts associated with other software in the same package even when they're called outside the initscript, wouldn't we be breaking this convention if we split them in this case?

shruthi-ravi replied:
> We always ship initscripts associated with other software in the same package even when they're called outside the initscript..

I didn't realize this was the case. I'm good with this then.

shruthi-ravi commented on recipes-ni/ni-utils/ni-utils_1.0.bb:
```diff
@@ -11,7 +11,10 @@ SRC_URI = "file://status_led \
 
 FILES_${PN} += "\
-	/usr/lib/nisetbootmode.functions \
+	${bindir}/status_led \
```
Shouldn't these LED changes be in the previous commit?

chaitu236 replied:
These changes aren't specifically for LED, I've replaced paths for all files in the package and it seemed better to do them together rather than breaking them up.

chaitu236 replied:
Moved them to previous commit as per offline conversation.

shruthi-ravi commented on recipes-ni/ni-utils/ni-utils_1.0.bb:
```diff
@@ -11,7 +11,10 @@ SRC_URI = "file://status_led \
-	/usr/lib/nisetbootmode.functions \
+	${bindir}/status_led \
+	${libdir}/nisetbootmode.functions \
+	${sysconfdir}/init.d/nisetbootmode \
```
Again, why move nisetbootmode here? Cleanup?

chaitu236 replied:
nisetbootmode is part of this package already (see below in do_install). I just added it to FILES section.
I'm not entirely sure why they're weren't already added to FILES section in first place, and why bitbake complains if nothing is added to FILES but doesn't when not all of them are.

amstewart commented on recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb:
```diff
@@ -14,8 +14,7 @@ RDEPENDS_${PN}-ptest += "bash"
 
-SRC_URI = "file://nisetbootmode \
-           file://firewall \
+SRC_URI = "file://firewall \
```
As long as you're touching this line, could you just refactor it like this, so that it stops showing up in the diff?

```
SRC_URI = " \
    file://firewall \
```

chaitu236 replied:
Done

amstewart reviewed (approved):
RE: `nisetbootmode` & `ni-utils`

I think we might need to refresh `nisetbootmode` to support RAUC. It seems like that script is intended for user use (LV API?) to trigger mode shifts. Certainly, the safemode/runmode code that is in there now will be meaningless in RAUC, and we'll need to add support for changing runlevels.

Strictly migrating it for now is fine, but we should create a work item to refresh the API and move it into its own package. As long as we're pulling it out of `ni-utils`, I don't really like that package's name - it's another "ni-misc", and we already have too many of those. So I think that work item should also ask to move the LED-management scripts into their own recipe/package. 

What do you think @jpautler ?

chaitu236 commented:
V2:
Addressed feedback.

Shruthi brought up the fact that nisetbootmode in meta-nilrt previously contained two files, one in bindir (that took auto-restore, restore, etc as params) and one in /etc/init.d (that has start and stop as params), so a user could invoke nisetbootmode with one of auto-restore, restore, etc without its full path as bindir is in PATH.
But I've now consolidated them into one file in /etc/init.d (like in os-common) so a user cannot invoke it without specifying the full path - but this isn't a concern because there doesn't seem to be any users that invoke the file directly (as per codesearch).

shruthi-ravi reviewed (approved):
Looks good, thanks for making the changes.

chaitu236 commented:
> RE: `nisetbootmode` & `ni-utils`
> 
> I think we might need to refresh `nisetbootmode` to support RAUC. It seems like that script is intended for user use (LV API?) to trigger mode shifts. Certainly, the safemode/runmode code that is in there now will be meaningless in RAUC, and we'll need to add support for changing runlevels.
> 
> Strictly migrating it for now is fine, but we should create a work item to refresh the API and move it into its own package. As long as we're pulling it out of `ni-utils`, I don't really like that package's name - it's another "ni-misc", and we already have too many of those. So I think that work item should also ask to move the LED-management scripts into their own recipe/package.
> 
> What do you think @jpautler ?

Agreed we need to update many scripts to remove safemode/runmode assumptions and replace with RAUC. Does [this](https://ni.visualstudio.com/DevCentral/_workitems/edit/1200346/?workitem=1198511) seem like a good feature to put the stories under?

amstewart commented:
That's the RAUC epic, so yes? I'm not sure what Feature it should go under though.
